SrfGen/srfinfo2vm.py

```diff
@@ -264,66 +264,33 @@ def save_vm_config(
-                        'sim_duration = "%s"' % (sim_duration),
-                        'flo = "%s"' % (min_vs / (5.0 * hh)),
-                        'nx = "%s"' % (int(round(float(xlen) / hh))),
-                        'ny = "%s"' % (int(round(float(ylen) / hh))),
-                        'nz = "%s"' % (int(round(float(zmax - zmin) / hh))),
-                        'sufx = "_%s01-h%.3f"' % (code, hh),
-                    ]
-                )
-            )
-        with open("%s.json" % (params_vel), "w") as pv:
-            # must also convert mag from np.float to float
-            json.dump(
-                {
-                    "mag": float(mag),
-                    "centroidDepth": centroid_depth,
-                    "MODEL_LAT": origin[1],
-                    "MODEL_LON": origin[0],
-                    "MODEL_ROT": rot,
-                    "hh": hh,
-                    "min_vs": min_vs,
-                    "model_version": model_version,
-                    "topo_type": topo_type,
-                    "output_directory": os.path.basename(vm_dir),
-                    "extracted_slice_parameters_directory": "SliceParametersNZ/SliceParametersExtracted.txt",
-                    "code": code,
-                    "extent_x": xlen,
-                    "extent_y": ylen,
-                    "extent_zmax": zmax,
-                    "extent_zmin": zmin,
-                    "sim_duration": sim_duration,
-                    "flo": min_vs / (5.0 * hh),
-                    "nx": int(round(float(xlen) / hh)),
-                    "ny": int(round(float(ylen) / hh)),
-                    "nz": int(round(float(zmax - zmin) / hh)),
-                    "sufx": "_%s01-h%.3f" % (code, hh),
-                },
-                pv,
-            )
-
+        # must also convert mag from np.float to float
+       dump_yaml(
+            {
+                "mag": float(mag),
+                "centroidDepth": float(centroid_depth),
+                "MODEL_LAT": float(origin[1]),
+                "MODEL_LON": float(origin[0]),
+                "MODEL_ROT": float(rot),
+                "hh": hh,
+                "min_vs": float(min_vs),
+                "model_version": model_version,
+                "topo_type": topo_type,
+                "output_directory": os.path.basename(vm_dir),
+                "extracted_slice_parameters_directory": "SliceParametersNZ/SliceParametersExtracted.txt",
+                "code": code,
+                "extent_x": float(xlen),
+                "extent_y": float(ylen),
+                "extent_zmax": float(zmax),
+                "extent_zmin": float(zmin),
+                "sim_duration": float(sim_duration),
+                "flo": min_vs / (5.0 * hh),
+                "nx": int(round(float(xlen) / hh)),
+                "ny": int(round(float(ylen) / hh)),
+                "nz": int(round(float(zmax - zmin) / hh)),
+                "sufx": "_%s01-h%.3f" % (code, hh),
+            },"%s.yaml" %(params_vel)
```

Comment:
I believe the % approach is obsolete and may be removed in the future, so I fan of switching to .format

https://docs.python.org/release/3.1.5/library/stdtypes.html#old-string-formatting-operations